### ISSUE TITLE
Issue on configuration steps for Jave language extension in SQL Server 2019

### ISSUE THREAD

**Issue by AnyiYang** [MicrosoftDocs/sql-docs#1669]
We totally follow the steps in the following document:

https://docs.microsoft.com/en-us/sql/advanced-analytics/java/java-first-sample?view=sqlallproducts-allversions

But failed to run the sp_execute_external_script sample in the document successfully under both jdk11.0.1 and jdk 10.0.2. The error message is shown as the following:

Msg 39004, Level 16, State 20, Line 1
A 'Java' script error occurred during execution of 'sp_execute_external_script' with HRESULT 0x80004004.

However, if I changed the service account for Launchpad service from default account NT Service\MSSQLLaunchpad to local admin, I can run the stored procedure and get the result successfully on both jdk environment.

It seems that the issue should be on permission. However, I have granted corresponding permission on the files for SQLRUserGroup and ALL Application Package based on the official documents. From the process monitor, I found access denied entries on both successful operation with local admin and the failed operation. So I failed to do the troubleshooting through process monitor.

Please help to check what other permissions are in need for the service account of SQL Server Launchpad for java

**Comment by VanMSFT** [comment]
Hi @AnyiYang, thanks for bringing this issue up to us. 

@HeidiSteen - Do you mind checking if there are any other permissions needed?

**Comment by HeidiSteen** [comment]
@AnyiYang -- Are you using Java 8?   I don't think these are Java 8:  jdk11.0.1 and jdk 10.0.2

**Comment by HeidiSteen** [comment]
@AnyiYang -- I cannot repro the issue, but since I just did this, I can see how it easy it might be to miss a step. These steps are documented, but I'll call them out here anyway.   Also please checkout item 5 -- it's the last one listed, but after re-reading your issue, it's probably the source of the problem.

1 - my java files are in a folder close to the root (I used "c:\javasample").
2 - permissions on c:\javasample and c:\javasample\pkgs are granted to "SQLRUserGroup" and "ALL APPLICATION PACKAGES"
3 - On SQL Server, I created a test database called 'test-java'. I ran the SQL script that creates the data in test-java.  Later, after all of the java work was done, I made sure to select test-java when executing SPEES (sp_execute_external_script).
4 - I had to add the Java BIN folder to my path.  JAVA_HOME by itself didn't cut it. The javac exe was "not recognized" unless BIN was in the path.
5 - People need permission to run SPEES.  Non-admin users are granted access as described in this article:  https://docs.microsoft.com/en-us/sql/advanced-analytics/security/user-permission?view=sql-server-ver15 

Since I can't repro, I have to recommend that we close this issue.  If you disagree or find something else, please let me know.

**Comment by VipinCMathew** [comment]
I have installed SQL Server Express Edition 2019 with Language Extensions on Windows machine.
Java version - Default java version used as part of the installation of SQL Server 2019 (AZUL-OpenJDK-JRE)

We have added the external languages and Libraries bases on the link
https://docs.microsoft.com/en-us/sql/t-sql/statements/create-external-language-transact-sql?view=sql-server-ver15

We tried to execute the sample code based on the below link
https://docs.microsoft.com/en-us/sql/language-extensions/tutorials/search-for-string-using-regular-expressions-in-java?view=sqlallproducts-allversions

While executing the script
EXEC sp_execute_external_script
  @language = N'Java'
, @script = N'pkg.RegexSample'
, @input_data_1 = N'SELECT id, text FROM testdata'
, @params = N'@regexExpr nvarchar(200)'
, @regexExpr = N'[Jj]ava'
with result sets ((ID int, text nvarchar(100)));

getting the following error.

**Msg 39004, Level 16, State 19, Line 2
A 'Java' script error occurred during execution of 'sp_execute_external_script' with HRESULT 0x80072746.**

Completion time: 2020-09-23T15:28:17.9396201+05:30

In order to solve this issue we referred this link https://github.com/MicrosoftDocs/sql-docs/issues/1669
We have given full permissions on the folder as per the link

We have logged in as user with sysadmin role.

Still we are getting the same Error.
